internal/watcher/watcher.go

```diff
@@ -18,9 +18,11 @@ var (
 	ErrBucketMismatch = errors.New("bucket mismatch")
 )
 
+type Cleanup func() error
```

Comment:
I would suggest one more change in the signature of this function:

    type Cleanup func(context.Context) error

By accepting a context with give the user the control to establish new deadlines or cancellation management for the cleanup routine specifically. Implicitly using the context provided elsewhere is limiting and uncommon.

Reply:
I was about to abject but I noticed that LREM takes a context. Yeah that is weird thanks for catching this.